lib/Wikia/src/CircuitBreaker/MemcachedCircuitBreakerStorage.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace Wikia\CircuitBreaker;
+
+use Ackintosh\Ganesha;
+use Ackintosh\Ganesha\Builder;
+use Ackintosh\Ganesha\Storage\Adapter\Memcached;
+use Wikia\Logger\Loggable;
+
+class MemcachedCircuitBreakerStorage implements CircuitBreakerStorage {
+
+	use Loggable;
+
+	/** @var Ganesha */
+	private $ganesha;
+
+	private function configure() {
+		global $wgMemCachedTimeout, $wgMemCachedConnectionTimeout;
+
+		return [
+			\Memcached::OPT_SEND_TIMEOUT => $wgMemCachedTimeout,
+			\Memcached::OPT_RECV_TIMEOUT => $wgMemCachedTimeout,
+			\Memcached::OPT_CONNECT_TIMEOUT => $wgMemCachedConnectionTimeout ?: 0.5
+		];
+	}
+
+	public function __construct() {
+		global $wgMemCachedServers;
+
+		$memCached = new \Memcached( );
+		$memCached->setOptions( $this->configure() );
+
+		foreach ( $wgMemCachedServers as $weight => $memCachedServer ) {
+			$parts = preg_split('/:/', $memCachedServer, 2);
```

Comment:
Minor: coding conventions:
```suggestion
			$parts = preg_split( '/:/', $memCachedServer, 2 );
```